app/models/task.py

```diff
@@ -1,5 +1,44 @@
 from app import db
+from datetime import datetime
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        return dict(
+            id = self.task_id,
+            title = self.title,
+            description = self.description,
+            is_complete = False
+        )
+
+    @classmethod
+    def from_dict(cls, task_data):
+        return cls(
+            title = task_data["title"],
+            description = task_data["description"],
+            completed_at = None
+        )
+
+    def true_dict(self):
+        return dict(
+            id = self.task_id,
+            title = self.title,
+            description = self.description,
+            is_complete = True
+        )
+    
+    def task_dict(self):
+        return dict(
+            id = self.task_id,
+            goal_id = self.goal_id,
+            title = self.title,
+            description = self.description,
+            is_complete = False
```

Comment:
You've got three methods in this class that returns a dictionary representing a Task, but they're all slightly different. It can be hard to keep track of the slight variations in these 3 methods and there is a bit of repetition. 

You could refactor to_dict() by adding in some conditional logic to set goal_id and is_complete when needed. Doing so would allow you to delete true_dict and task_dict